clash-lib/src/Clash/Normalize/Transformations/Specialize.hs

```diff
@@ -488,6 +500,18 @@ specialize' (TransformContext is0 _) e (Var f, args, ticks) specArgIn = do
     noUserInline = NoUserInline
 #endif
 
+    specializeName :: (InlineSpec, Name Term) -> (Maybe InlineSpec, Name Term) -> Name Term
+    specializeName (spec0, n0) (Just spec1, n1)
+      | isNoInline spec0
+      , isNoInline spec1
+      = n0{nameOcc=nameOcc n0 <> "_" <> stripMods (nameOcc n1)}
+      | isNoInline spec1
+      = n1
+    specializeName (_, n0) _ = n0
```

Comment:
This implement:

| OPAQUE | Name for `f g` |
|-|-|
| f and g | `f_g` |
| otherwise | g |

And thus does not implement what's in the comments at https://github.com/clash-lang/clash-compiler/pull/2509/files#diff-38424dec2448cd00ad74cbd954d63cd04bc9d70ae89eaf7eecec8087669d2a33R470-R475

Reply:
In addition, I thought we all agreed that we would like to have:

| OPAQUE | Name for `f g` |
| - | - |
| f and g | `f_g` |
| f | `f` |
| g | `g` | 
| otherwise | `f_g` |

Reply:
We've got OPAQUE f and g:

```
      | isNoInline spec0
      , isNoInline spec1
      = n0{nameOcc=nameOcc n0 <> "_" <> stripMods (nameOcc n1)}
```

resulting in `f_g`. We've got just OPAQUE `g`, which does:

```
      | isNoInline spec1
      = n1
```

I.e., just `g`. Then for everything else we pick `f`:

```
    specializeName (_, n0) _ = n0
```


This wasn't my intention though, I wanted to write:

```
      | isNoInline spec0
      = n0
    specializeName (_, n1) _ = n1
```

..to match the comment.

Reply:
> In addition, I thought we all agreed that we would like to have:

Yes, though this code predates that discussion :-)